cloud/services/compute/loadbalancers/reconcile.go

```diff
@@ -18,72 +18,240 @@ package loadbalancers
 	if err != nil {
 		return err
 	}
 
-	healthcheck, err := s.createOrGetHealthCheck(ctx)
+	lbSpec := s.scope.LoadBalancer()
+	lbType := ptr.Deref(lbSpec.LoadBalancerType, infrav1.ExternalLoadBalancerOnly)
+	// Create a Global External Proxy Load Balancer by default
+	if lbType == infrav1.ExternalLoadBalancerOnly || lbType == infrav1.DualLoadBalancer {
+		if err = s.createExternalLoadBalancer(ctx, lbType, instancegroups); err != nil {
+			return err
+		}
+	}
+
+	// Create a Regional Internal Passthrough Load Balancer if configured
+	if lbType == infrav1.DualLoadBalancer || lbType == infrav1.InternalLoadBalancerOnly {
+		name := infrav1.InternalRoleTagValue
+		if lbSpec.InternalLoadBalancer != nil {
+			name = ptr.Deref(lbSpec.InternalLoadBalancer.Name, infrav1.InternalRoleTagValue)
+		}
+		if err = s.createInternalLoadBalancer(ctx, name, lbType, instancegroups); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Delete deletes cluster control-plane loadbalancer components.
+func (s *Service) Delete(ctx context.Context) error {
+	log := log.FromContext(ctx)
+	var externalLBErr, internalLBErr error
+	lbSpec := s.scope.LoadBalancer()
+	lbType := ptr.Deref(lbSpec.LoadBalancerType, infrav1.ExternalLoadBalancerOnly)
+	if lbType == infrav1.ExternalLoadBalancerOnly || lbType == infrav1.DualLoadBalancer {
+		externalLBErr = s.deleteExternalLoadBalancer(ctx)
+	}
+
+	if lbType == infrav1.DualLoadBalancer || lbType == infrav1.InternalLoadBalancerOnly {
+		name := infrav1.InternalRoleTagValue
+		if lbSpec.InternalLoadBalancer != nil {
+			name = ptr.Deref(lbSpec.InternalLoadBalancer.Name, infrav1.InternalRoleTagValue)
+		}
+		internalLBErr = s.deleteInternalLoadBalancer(ctx, name)
+	}
+	if externalLBErr != nil {
+		if err := s.deleteInstanceGroups(ctx); err != nil {
+			log.Error(err, "Error deleting instancegroup")
+		}
+		return externalLBErr
+	}
+	if internalLBErr != nil {
+		if err := s.deleteInstanceGroups(ctx); err != nil {
+			log.Error(err, "Error deleting instancegroup")
+		}
+		return internalLBErr
+	}
+
+	// Delete shared instancegroups last
+	return s.deleteInstanceGroups(ctx)
+}
+
+func (s *Service) deleteExternalLoadBalancer(ctx context.Context) error {
+	log := log.FromContext(ctx)
+	log.Info("Deleting external loadbalancer resources")
+	name := infrav1.APIServerRoleTagValue
+	if err := s.deleteForwardingRule(ctx, name); err != nil {
+		return err
+	}
+	s.scope.Network().APIServerForwardingRule = nil
+
+	if err := s.deleteAddress(ctx, name); err != nil {
+		return err
+	}
+	s.scope.Network().APIServerAddress = nil
+
+	if err := s.deleteTargetTCPProxy(ctx); err != nil {
+		return err
+	}
+	s.scope.Network().APIServerTargetProxy = nil
+
+	if err := s.deleteBackendService(ctx, name); err != nil {
+		return err
+	}
+	s.scope.Network().APIServerBackendService = nil
+
+	if err := s.deleteHealthCheck(ctx, name); err != nil {
+		return err
+	}
+	s.scope.Network().APIServerHealthCheck = nil
```

Comment:
I'd add an `fmt.Errorf(` wrap for all these errors for easier debugging when following the log error chain

Reply:
Done.